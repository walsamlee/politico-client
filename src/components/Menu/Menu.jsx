/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import { Link } from 'react-router-dom';
import Logout from '../Logout/Logout';
import getToken from '../../utils/getToken';
import getPrivilege from '../../utils/getPrivilege';

const Menu = () => {
  const token = getToken();
  const privilege = getPrivilege();

  if (token) {
    if (privilege === 'true') {
      return (
        <div>
          <nav>
            <div className="pry-menu-items">
              <Link to="/parties">Parties</Link>
              <Link to="/create-party">Create Party</Link>
              <Link to="/create-office">Create Office</Link>
            </div>
            <div className="logout">
              <Logout />
            </div>
          </nav>
          <div className="mini-menu-icon">
            <img src="https://res.cloudinary.com/walsam/image/upload/v1559199507/politico-mini.png" alt="logo" />
            <label htmlFor="checker" className="checker-label">
              <p><i className="fas fa-bars" /></p>
            </label>
          </div>
        </div>
      );
    }
    return (
      <div>
        <nav>
          <div className="pry-menu-items">
            <Link to="/parties">Parties</Link>
            <Link to="/candidates">Vote</Link>
            <Link to="/history">Voting History</Link>
          </div>
          <div className="logout">
            <Logout />
          </div>
        </nav>
        <div className="mini-menu-icon">
          <img src="https://res.cloudinary.com/walsam/image/upload/v1559199507/politico-mini.png" alt="logo" />
          <label htmlFor="checker">
            <p><i className="fas fa-bars" /></p>
          </label>
        </div>
      </div>
    );
  }
  return (
    <div>
      <nav>
        <div className="pry-menu-items">
          <Link to="/parties">Parties</Link>
        </div>
        <div className="logout">
          <Link to="/login" id="login">Login</Link>
          <Link to="/register" id="signup">Register</Link>
        </div>
      </nav>
      <div className="mini-menu-icon">
        <img src="https://res.cloudinary.com/walsam/image/upload/v1559199507/politico-mini.png" alt="logo" />
        <label htmlFor="checker">
          <p><i className="fas fa-bars" /></p>
        </label>
      </div>
    </div>
  );
};

export default Menu;
