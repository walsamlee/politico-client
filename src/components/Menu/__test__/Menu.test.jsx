import React from 'react';
import { shallow } from 'enzyme';
import Menu from '../Menu';

describe('<Menu />', () => {
  it('should render Menu component', () => {
    const wrapper = shallow(
      <Menu />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'));
  });
});
