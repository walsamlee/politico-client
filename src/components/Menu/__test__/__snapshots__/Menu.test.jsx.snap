// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render Menu component 1`] = `ShallowWrapper {}`;
